fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/data/LoanSummaryData.java

```diff
@@ -96,7 +96,7 @@ public LoanSummaryData(final CurrencyData currency, final BigDecimal principalDi
         this.interestOverdue = interestOverdue;
         this.feeChargesCharged = feeChargesCharged;
         this.feeChargesDueAtDisbursementCharged = feeChargesDueAtDisbursementCharged;
-        this.feeChargesPaid = feeChargesPaid;
+        this.feeChargesPaid = feeChargesPaid != null ? feeChargesPaid : BigDecimal.ZERO;
```

Comment:
For generic behaviour, i would not set only the feeChargesPaid to Zero.

Reply:
Well spotted. I checked if the reason for this line was outside of this constructor, but it's used only in one place. The Git history doesn't show any use of "ZERO", will revert that line.